src/Enemy.cpp

```diff
@@ -14,8 +14,94 @@ int Enemy::getTextureWidth(){
+        0,
+        nullptr
+    );
+    
+    fringe.push(startNode);
+    
+    bool found = false;
+    PathFindingNode *finalNode = nullptr;
+    while(!found){
+        if (fringe.empty()){
+            printf("Could not find target\n");
+            return;
+        }
+
+        PathFindingNode *currentNode = fringe.top();
+        fringe.pop();
+
+        if(currentNode->getTileX() == targetTileX && currentNode->getTileY() == targetTileY){
+            finalNode = currentNode;
+            found = true;
+        } else {
+            addNode(currentNode->getTileX() + 1, currentNode->getTileY(), targetTileX, targetTileY, currentNode->getGCost() + 1, currentNode, fringe);
+            addNode(currentNode->getTileX() - 1, currentNode->getTileY(), targetTileX, targetTileY, currentNode->getGCost() + 1, currentNode, fringe);
+            addNode(currentNode->getTileX(), currentNode->getTileY() + 1, targetTileX, targetTileY, currentNode->getGCost() + 1, currentNode, fringe);
+            addNode(currentNode->getTileX(), currentNode->getTileY() - 1, targetTileX, targetTileY, currentNode->getGCost() + 1, currentNode, fringe);
+            addNode(currentNode->getTileX() + 1, currentNode->getTileY() + 1, targetTileX, targetTileY, currentNode->getGCost() + sqrt(2), currentNode, fringe);
+            addNode(currentNode->getTileX() + 1, currentNode->getTileY() - 1, targetTileX, targetTileY, currentNode->getGCost() + sqrt(2), currentNode, fringe);
+            addNode(currentNode->getTileX() - 1, currentNode->getTileY() + 1, targetTileX, targetTileY, currentNode->getGCost() + sqrt(2), currentNode, fringe);
+            addNode(currentNode->getTileX() - 1, currentNode->getTileY() - 1, targetTileX, targetTileY, currentNode->getGCost() + sqrt(2), currentNode, fringe);
+        }
+    }
+    
+    float rot;
+    if(startNode->getTileX() == finalNode->getTileX() && startNode->getTileY() == finalNode->getTileY()){
+        rot = M_PI - atan2(tx - this->getPosition().x, ty - this->getPosition().y);
+    } else {
+        while(finalNode->getPreviousNode()->getTileX() != startNode->getTileX() || finalNode->getPreviousNode()->getTileY() != startNode->getTileY()){
+            PathFindingNode *temp = finalNode;
+            finalNode = finalNode->getPreviousNode();
+            delete temp;
+        }
+        float scaleBacktoPosition = MainView::getInstance().getSize().x / GameConfig::getInstance().getJson("resources/configs/const-settings.json").at("map").at("tiles-per-window-width").as_double();
+        
+        float positionX = this->getPosition().x;
+        float positionY = this->getPosition().y;
+
+        // negative coords need position to be scaled back one tile for trig function to work properly
+        if (finalNode->getTileX() < 0) {
+            positionX += scaleBacktoPosition;
+        }
+
+        if (finalNode->getTileY() < 0) {
+            positionY += scaleBacktoPosition;
+        }
```

Comment:
This was the negative coords issue fix. Kinda weird but the math just makes sense